src/schemaGenerator.ts

```diff
@@ -460,6 +463,24 @@ function addResolveFunctionsToSchema(
       }
     });
   });
+
+  // If we have any union or interface types throw if no there is no resolveType or isTypeOf resolvers
+  if (requireResolversForResolveType) {
+    checkForResolveTypeResolver(schema);
```

Comment:
What if we inverted the option so that a missing `resolveType` caused a warning, but you could pass `resolverValidationOptions.allowMissingResolveType` (or something like that) to disable the warning?

I'm worried no one is going to know they should start passing `requireResolversForResolveType: true`, so they won't ever see this warning.